src/esp/assets/ResourceManager.cpp

```diff
@@ -422,14 +433,76 @@ esp::geo::CoordinateFrame ResourceManager::buildFrameFromAttributes(
   }
 }  // ResourceManager::buildCoordFrameFromAttribVals
 
+bool ResourceManager::loadRenderAsset(const AssetInfo& info) {
+  bool meshSuccess = false;
+  if (info.type == AssetType::FRL_PTEX_MESH) {
+    meshSuccess = loadRenderAssetPTex(info);
+  } else if (info.type == AssetType::INSTANCE_MESH) {
+    meshSuccess = loadRenderAssetIMesh(info);
+  } else if (isRenderAssetGeneral(info.type)) {
+    meshSuccess = loadRenderAssetGeneral(info);
+  } else {
+    ASSERT(
+        false);  // loadRenderAsset doesn't yet support the requested asset type
+  }
+#if 0  // coming soon
+  if (renderKeyframeWriter_) {
+    renderKeyframeWriter_->onLoadRenderAsset(info);
+  }
+#endif
+  return meshSuccess;
+}
+
+scene::SceneNode* ResourceManager::createRenderAssetInstance(
+    const RenderAssetInstanceCreation& creation,
+    scene::SceneNode* parent,
+    DrawableGroup* drawables,
+    std::vector<scene::SceneNode*>* visNodeCache) {
+  // assert that asset is already loaded
+  CHECK(resourceDict_.count(creation.filepath));
+
+  const LoadedAssetData& loadedAssetData = resourceDict_.at(creation.filepath);
+  if (!isLightSetupCompatible(loadedAssetData, creation.lightSetupKey)) {
+    LOG(WARNING)
+        << "Instantiating render asset " << creation.filepath
+        << " with incompatible light setup, instance will not be correctly lit."
+           "For objects, please ensure 'requires lighting' is enabled in "
+           "object config file.";
+  }
+
+  const auto& info = loadedAssetData.assetInfo;
+  scene::SceneNode* newNode = nullptr;
+  if (info.type == AssetType::FRL_PTEX_MESH) {
+    ASSERT(
+        !visNodeCache);  // createRenderAssetInstancePTex doesn't support this
+    newNode = createRenderAssetInstancePTex(creation, parent, drawables);
+  } else if (info.type == AssetType::INSTANCE_MESH) {
+    ASSERT(
+        !visNodeCache);  // createRenderAssetInstanceIMesh doesn't support this
+    newNode = createRenderAssetInstanceIMesh(creation, parent, drawables);
+  } else if (isRenderAssetGeneral(info.type) ||
+             info.type == AssetType::PRIMITIVE) {
+    newNode = createRenderAssetInstanceGeneralPrimitive(
+        creation, parent, drawables, visNodeCache);
+  } else {
+    ASSERT(false);  // createRenderAssetInstance doesn't yet support the
+                    // requested asset type
+    newNode = nullptr;
```

Comment:
`CORRADE_INTERNAL_ASSERT_UNREACHABLE()` here again, and because it marks the code as unreachable, the `newNode = nullptr` isn't needed to suppress "potentially uninitialized variable" warnings.